src/libexpr/primops/fromYAML.cc

```diff
@@ -0,0 +1,155 @@
+// RYML: workaround for bug in amalgamate script - nix requires at least C++17
+#include <charconv>
+// enabling this macro has a performance penalty
+#define RYML_WITH_TAB_TOKENS
+#define RYML_SINGLE_HDR_DEFINE_NOW
+#include "../../rapidyaml/ryml_all.hpp"
+
+#include "primops.hh"
+#include "eval-inline.hh"
+
+
+namespace nix {
+
+struct NixContext {
+    EvalState & state;
+    const PosIdx pos;
+    std::string_view yaml;
+};
+
+static void s_error [[ noreturn ]] (const char* msg, size_t len, ryml::Location loc, void *nixContext)
+{
+    auto context = (const NixContext *) nixContext;
```

Comment:
You shouldn't look at other parts of nix... In this case `static_cast` is not really safer than a C-type cast because `void*` has not much type information. So it's more a matter of convention, but of course I can apply your suggestion.

Reply:
Of course i have no authority so define what's right if there is enough historic precedence of the C-way but it also looks like it gets worse if we just stay at it. It would be awesome to gradually get back to C++